### ISSUE TITLE
cannot pull from docker registry with self-signed certificate

### ISSUE THREAD

**Issue by JohnDoe2991** [microsoft/vscode-remote-release#5276]
- VSCode Version: 1.57.1
- Local OS Version: Windows 10 2004
- Remote OS Version: Docker Image based on Ubuntu 20.04
- Remote Extension/Connection Type: Docker
- Logs:
Ouput after "Rebuild container" (I removed all personal information from this output):
```
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 38B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1.2               3.2s
 => CACHED docker-image://docker.io/docker/dockerfile:1.2@sha256:e2a8561e  0.0s
 => ERROR [internal] load metadata for docker.privatregistry.local/librar  21.1s
------
 > [internal] load metadata for docker.privatregistry.local/library/testimage:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gatew
ay.v0: rpc error: code = Unknown desc = failed to do request: Head https://docke
r.privatregistry.local/v2/library/testimage/manifests/1.0.0: dial tc
p xxx.xxx.xxx.xxx:443: connect: connection refused
[30449 ms] Command failed: docker build -f /path/to/workspace/.devcontainer/Dockerfile -t container_name --build-arg USERNAME=username /path/to/workspace/.devcontainer
```
Steps to Reproduce:

1. create a private registry (e.g. Habor) with a self-signed certificate
2. add self-signed certificate to Windows, Docker for Windows and WSL2
3. push an testimage to the registry
4. create a workspace with a devcontainer.json and Dockerfile based on the testimage
5. let VSCode rebuild and open the devcontainer

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No
I can push and pull manually within Powershell or WSL2 without any problems, only VSCode seems to be unable to access the registry. If I pull the testimage manually in Powershell and let VSCode then build, everything works because it uses the cached image, I think.

**Comment by chrmarti** [comment]
We are running `docker build` either on Windows or on WSL (when opening a WSL folder), that should use certificates the same way it does when you run it from the command line. Have you checked https://docs.docker.com/registry/insecure/#use-self-signed-certificates?

**Comment by JohnDoe2991** [comment]
I think I've resolved the issue with your documentation:

I've added the certificate to Windows and to the Docker for Desktop client (C:\ProgramData\Docker\certs.d\REGISTRY_URL\).
I can push/pull within Powershell, Ubuntu console (WSL2) and VSCode consoles (in Windows and WSL2) without any issue. But VSCode itself is not able to build the devcontainer.

Then I've added the certificate to Ubuntu (WSL2). Now VSCode is also able to build the devcontainer.

Thank you for your help!
